main.js

```diff
@@ -56,38 +59,226 @@ function setupInitialPage() {
   renderContent(initialUrl);
 }
 
-function renderHomePage() {
+async function renderHomePage() {
   const mainContentDiv = document.querySelector('.main-content-component');
   mainContentDiv.innerHTML = getHomePageTemplate();
-  // Sample hardcoded event data
-  const eventData = {
-    id: 1,
-    description: 'Sample event description.',
-    img: 'https://images.unsplash.com/photo-1540575467063-178a50c2df87?ixlib=rb-4.0.3&ixid=M3wxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8fA%3D%3D&auto=format&fit=crop&w=1170&q=80',
-    name: 'Sample Event',
-    ticketCategories: [
-      { id: 1, description: 'General Admission' },
-      { id: 2, description: 'VIP' },
-    ],
-  };
-  // Create the event card element
-  const eventCard = document.createElement('div');
-  eventCard.classList.add('event-card'); 
-  // Create the event content markup
+
+  console.log('function', fetchTicketEvents());
+  fetchTicketEvents().then((data) => {
+    addEvents(data);
+  });
+}
+
+function addImage(name){
+  var imagesrc;
+  if(name =="Untold"){
+    imagesrc ="https://viacluj.tv/wp-content/uploads/2022/08/untold-3.jpg";
+  }
+  else
+  if(name =="Electric Castle"){
+    imagesrc ="https://image.stirileprotv.ro/media/images/original/Jul2023/62367524.jpg";
+  }
+  else
+  if(name =="Meci de fotbal"){
+    imagesrc ="https://i.eurosport.com/2022/10/19/3475646-70860088-2560-1440.jpg";
+  }
+  else
+  if(name =="Wine Festival"){
+    imagesrc ="https://zcj.ro/images/db/1_3_250125_1688475563_06498.jpg"; 
+  }
+  return imagesrc;
+}
+
+async function fetchTicketEvents(){
+  const response = await fetch('http://localhost:8082/api/allEvents',{mode:'cors'});
+  const data = await response.json();
+  console.log(data);
+  return data;
+}
+
+const addEvents = (events) =>{
+  const eventDiv = document.querySelector('.events');
+  eventDiv.innerHTML = "No ivents";
+  if(events.length){
+    eventDiv.innerHTML = "";
+    events.forEach(event =>{
+      eventDiv.appendChild(createEvent(event));
+    });
+  }
+};
+
+const createEvent = (eventData) =>{
+  //const title = kebabCase(eventData.eventType.name);
+  const title = kebabCase(eventData.name);
+  const eventElement = createEventElement(eventData, title);
+  return eventElement;
+};
+
+const createEventElement = (eventData, title) =>{
```

Comment:
This function is a bit too big.
Separate a bit the concerns and create smaller functions , to handle the quantity changes, event type changes, etc,